sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/fcm/RenewalWorkerDataContainer.java

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.android.communication.chat.implementation.notifications.fcm;
+
+import java.security.NoSuchAlgorithmException;
+
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+
+import lombok.Getter;
+import lombok.Setter;
+
+/**
+ * A singleton wrapper which contains KeyGenerator, secret keys and execution result related to #{RenewalTokenWorker}.
+ * The major reason for having this class is that Peoriodic Worker only has ENQUEUED and RUNNING state. There is no
+ * approach to pass execution result data by design. We use this shared instance to maintain values that we need by
+ * both PushNotificationClient and RenewalTokenWorker.
+ */
+public class RenewalWorkerDataContainer {
+    private static RenewalWorkerDataContainer renewalWorkerDataContainer;
+
+    private KeyGenerator keyGenerator;
+
+    @Getter
+    @Setter
+    private SecretKey curCryptoKey;
+
+    @Getter
+    @Setter
+    private SecretKey curAuthKey;
+
+    @Getter
+    @Setter
+    private SecretKey preCryptoKey;
+
+    @Getter
+    @Setter
+    private SecretKey preAuthKey;
+
+    @Getter
+    @Setter
+    private boolean executionFail;
+
+    @Getter
+    @Setter
+    private long keyRotationTime;
+
+    public RenewalWorkerDataContainer()  {
+        try {
+            this.keyGenerator = KeyGenerator.getInstance("AES");
+        } catch (NoSuchAlgorithmException e) {
+            e.printStackTrace();
```

Comment:
We should use [`ClientLogger.logExceptionAsError()`](https://github.com/Azure/azure-sdk-for-android/blob/9fc5dbc53de69a20d9009a5bad9cd9ff0392ff54/sdk/core/azure-core-logging/src/main/java/com/azure/android/core/logging/ClientLogger.java#L206) instead.